Source/WTF/wtf/NativePromise.h

```diff
@@ -957,12 +978,21 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
         using ThenCallbackType = ThenCallback<IsChaining::value, decltype(std::declval<LambdaReturnTrait>().template type<R1>())>;
         using ReturnType = ThenCommand<ThenCallbackType>;
 
-        auto thenCallback = adoptRef(*new ThenCallbackType(RefPtr { &targetQueue }, WTFMove(lambda), callSite));
+        RefPtr<RefCountedSerialFunctionDispatcher> queue;
+        if constexpr (ShouldCallNativePromiseDispatcherGetter<Dispatcher>::value)
+            queue = targetQueue.nativePromiseDispatcher();
+        else if constexpr (IsSmartRef<Dispatcher>::value)
```

Comment:
Should check if it's a RefCountedSerialFunctionDispatcher and be only passed by reference 

Should use the code prior the introduction of RefCountedSerialFunctionDispatcher instead

Reply:
As discussed offline, I think we should only ever pass a `RefCountedSerialFunctionDispatcher` and so you call `whenSettled(scope.nativePromiseDispatcher(), ...)`